10_Examples/jett-onboarding/src/testFile.cpp

```diff
@@ -0,0 +1,43 @@
+#include "gtest/gtest.h"
+#include "talking.h"
+#include "string.h"
+
+class TestExample : public ::testing::Test
+{
+protected:
+  TestExample()
+  {
+    talking_ = std::make_shared<talking>();
+  }
+
+  void SetUp() override
+  {
+  }
+
+  void TearDown() override
+  {
+  }
+
+  std::shared_ptr<talking> talking_;
+
+};
+
+TEST_F(TestExample, testTalk) {
+  //testing the talk function
+  EXPECT_EQ("Test", talking_->talk("Test"));
+}
+
+TEST_F(TestExample, testNumberPrint) {
+  EXPECT_TRUE(talking_->numberPrint() == talking_->num1);
+}
+
+TEST_F(TestExample, testTwoToPower) {
+  EXPECT_EQ(talking_->twoToPower(5) == (2 ^ 5));
```

Comment:
EXPECT_EQ takes two arguments, so thinking this will fail. You could switch to EXPECT_TRUE, or remove equality and make it comma separated

Reply:
Another important thing to know:
EXPECT_EQ is fine for ints and bools, but floats and doubles can be fussy. Sometimes, they are basically the same number, but very very very slightly off due to machine precision. So for example, your equation might spit out 3.0000000000001 instead of 3.

For those, gtest has EXPECT_FLOAT_EQ, EXPECT_DOUBLE_EQ, and EXPECT_NEAR.